### PR TITLE
add cgroup function that can limit cpu share of each worker

### PR DESCRIPTION
**Pull request by caishize**
add cgroup function that can limit cpu share of each worker, and it really help indeed

### CODE REVIEW

File: storm-core/src/clj/backtype/storm/cgroup/cgroup_manager.clj [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        name (:name hierarchy)]
+    (if (util/exists-dir? dir)
+      (loop [a_hierarchy (get-hierarchies)]
+        (if-let [_hierarchy (first a_hierarchy)]
+          (if (and (= dir (:dir _hierarchy)) (= name (:name _hierarchy)))
+            _hierarchy
+            (recur (rest a_hierarchy))))
+        ))))
+
+;(defn mount [hierarchy]
```
Comment (lazyval) [inline storm-core/src/clj/backtype/storm/cgroup/cgroup_manager.clj]: commented code? why would you want it?

**Comment by lazyval** [comment]
AFAIK, if some feature is non-trivial, it requires backing JIRA ticket for it

File: storm-core/src/clj/backtype/storm/cgroup/cgroup_manager.clj (hunk 1 above)
Comment (caishize) [inline storm-core/src/clj/backtype/storm/cgroup/cgroup_manager.clj]: somebody may want to mount cgroup hierarchy in his code, not just with linux command.. i will delete it later

**Comment by harshach** [comment]
@caishize please file a JIRA here https://issues.apache.org/jira/browse/STORM . I haven't taken a close look at the changes but does any of these changes interfere with running on windows?

**Comment by revans2** [comment]
@caishize Any update on this?  I really would like to have cgroup support, and this should work really well with the resource aware scheduling we are working on, on another JIRA.

**Comment by caishize** [comment]
Yes,there's an update haven't been  submitted, but it still only works on an Linux operating system which has the  cgroup installed..

**Comment by jerrypeng** [comment]
any updates on this PR?

**Comment by hustfxj** [comment]
it is great that storm begin to support cgroup function just like jstorm

**Comment by d2r** [comment]
@revans2 @hustfxj @jerrypeng @caishize 

Should we add support for cgroups with the jstorm merge, or should we try to add support for it now?

@caishize It would be good to create a Jira issue for this change and resolve the conflicts with the master branch.

**Comment by ptgoetz** [comment]
I agree with @d2r, or would at least answer that, yes, we should hold off on cgroups support until the jstorm merge. I believe that @revans2 might have already covered this in JIRA. If not, please open a new issue.

**Comment by bastiliu** [comment]
@ptgoetz  Yes, @revans2  has created one. Following is the JIRA for jstorm merge.
https://issues.apache.org/jira/browse/STORM-1336

**Comment by knusbaum** [comment]
JStorm's cgroups merged with #1053. Closing this PR.

**Pull request closed without merging**